clippy_lints/src/if_let_redundant_pattern_matching.rs

```diff
@@ -74,11 +74,11 @@ impl<'a, 'tcx> LateLintPass<'a, 'tcx> for Pass {
                                    arms[0].pats[0].span,
                                    &format!("redundant pattern matching, consider using `{}`", good_method),
                                    |db| {
-                    let span = Span {
-                        lo: expr.span.lo,
-                        hi: op.span.hi,
-                        ctxt: expr.span.ctxt,
-                    };
+                    let span = Span::new(
+                        expr.span.lo(),
+                        op.span.hi(),
+                        expr.span.ctxt(),
+                    );
```

Comment:
Nit: this is `expr.span.with_hi(op.span.hi())`

Reply:
Thanks!